assets/src/components/AssignmentTable.js

```diff
@@ -120,170 +144,271 @@ function AssignmentTable (props) {
     }
   }, [currentWeekRow.current])
 
+  const [assignmentGroupNames, setAssignmentGroupNames] = useState([])
+  useEffect(() => {
+    const allGroupNames = assignments.map(ag => ag.assignmentGroup.name)
+    var uniqueGroupNames = [...new Set(allGroupNames)]
+    uniqueGroupNames.sort()
+    setAssignmentGroupNames(uniqueGroupNames)
+  }, [assignments])
+
+  const [assignmentFilter, setAssignmentFilter] = useState('')
+  const handleNameFilterChange = e => {
+    setAssignmentFilter(e.target.value)
+  }
+
+  const ITEM_HEIGHT = 48
+  const ITEM_PADDING_TOP = 8
+  const MenuProps = {
+    PaperProps: {
+      style: {
+        maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+        width: 250
+      }
+    }
+  }
+
+  const [assignmentGroupFilterArray, setAssignmentGroupFilterArray] = useState([])
+  const handleGroupNameArrayFilterChange = e => {
+    setAssignmentGroupFilterArray(e.target.value)
+  }
```

Comment:
I think this could be removed, and put inline on line 198: 
`onChange={e => setAssignmentGroupFilterArray(e.target.value)}`